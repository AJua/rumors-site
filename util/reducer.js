export const commonSetState = (state, { payload: { key, value } }) =>
  state.setIn(['state', key], value);
