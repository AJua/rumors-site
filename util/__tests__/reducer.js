import { fromJS } from 'immutable';
import { commonSetState } from '../reducer';

describe('reducer', () => {
  describe('commonSetState', () => {
    const initState = fromJS({
      state: {
        foo: 'bar',
      },
    });
    it('sets state from payload', () => {
      expect(
        commonSetState(initState, {
          payload: { key: 'foo', value: 'bar2' },
        }).get('state')
      ).toMatchSnapshot();

      expect(
        commonSetState(initState, {
          payload: { key: 'not-exist-key', value: 'new-value' },
        }).get('state')
      ).toMatchSnapshot();
    });
  });
});
