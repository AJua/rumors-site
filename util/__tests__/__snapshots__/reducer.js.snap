// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer commonSetState sets state from payload 1`] = `
Immutable.Map {
  "foo": "bar2",
}
`;

exports[`reducer commonSetState sets state from payload 2`] = `
Immutable.Map {
  "foo": "bar",
  "not-exist-key": "new-value",
}
`;
